idealalphaPackage.R

```diff
@@ -0,0 +1,201 @@
+  #f <- max(na.omit(info))
+  maxinf <- max(na.omit(info))
+  postd <- as.data.frame(subset(df, disconf > postdis))
+  postd <- min(postd$disconf)
+  d <- subset(df, disconf == postd)
+  #plot
+  plot(alpha,info, ylab = "Expected Change in Belief", xlab = "Signifcance Level", type = "l", main = paste("Alpha for",round(postd, digits = 2), "Probability after Non-Significance") , xlim = c(0,1), axes = F, ylim =c(0,round(maxinf + 0.005, digits = 2)))
+  abline(v = 0.05, col = "red")
+  abline(v = d$alpha, col = "blue")
+  axis(side=2, at=c(seq(0,(maxinf+0.01),0.01)))
+  axis(side=1, at=c(seq(0,1,0.05)))
+  mtext(side =3, paste("Alpha = ", round(d$alpha, digits = 2), " Power = ", round(d$power, digits = 2), "Learning =", round(d$info*100, digits = 2), "%"))
+  return(d)
+}
+
+#Probability based power analysis: Set both the posterior probability after a positive result and the posterior probability after a negative result
+#The function will calculate the required sample size and alplha level to achieve this.
+#' @export
+probpower <- function(effsize, prior, postdis, postconf, test = "t-test", type = "two.sample", alternative = "two.sided", k = 4)
+{
+
+  if(!postdis < prior | !prior < postconf)
+  {
+    stop("Posterior after disconfirming needs to be smaller than prior, posterior after confirming larger")
+  }
+  alpha <- seq(.0001,1,.001)
+  #for 10-20 000 particpants calculte the alpha needed to achieve the posterior after a negative result
+  #stop the loop when there is an option that at the same time has the needed posterior probability after a positive result
+  for (i in 10:20000) {
+  power <- get_power(alpha, effsize, i, test, type, alternative, k)
+  #calculate power for all alphas for this sample size
+  disconf <- postdisconfirm(power, alpha, prior)
+  confirm <- postconfirm(power, alpha, prior)
+  info <- information(confirm, disconf, power, alpha, prior)
+  maxinf <- max(na.omit(info))
+  df <- as.data.frame(cbind(alpha,power, disconf, confirm, info))
+  postd <- as.data.frame(subset(df, disconf > postdis))
+  postd <- min(postd$disconf)
+  d <- subset(df, disconf == postd)
+  d <- cbind(d, i)
+  postc <- d$confirm
+
+  if(postd > postdis & postc > postconf)
+  {
+    break
+  }
+  }
+  colnames(d) <- (c("alpha", "power", "disconf", "confirm", "info", "n"))
+  plot(df$alpha,df$info, ylab = "Expected Correct Change in Belief", xlab = "Signifcance Level", type = "l", main = paste("Probability bas Power Analysis - PosteriorConfirmation:",postconf, "  Posterior Disconfirmation:", postdis) , xlim = c(0,1), axes = F, ylim =c(0,round(maxinf + 0.005, digits = 2)))
+  abline(v = 0.05, col = "red")
+  abline(v = d$alpha, col = "blue")
+  axis(side=2, at=c(seq(0,(maxinf+0.01),0.01)))
+  axis(side=1, at=c(seq(0,1,0.05)))
+  mtext(side =3, paste("Sample Size", i, " Alpha = ", round(d$alpha, digits = 2), " Power = ", round(d$power, digits = 2), "Learning =", round(d$info*100, digits = 2), "%"))
+  return(d)
+}
```

Comment:
It is quite hard to find anything to really point your eyes towards, because you write such consistent text with consistent functions. It is really nice to see. For newly acquired taste during cleaning, some of your objects feel sometimes a bit short for your main script and would rather fit in addition. However, I can totally relate and do it myself. On the oother hand, you can easily  replace abbreviation that you once used inside RStudio, so it might be something to consider. Otherwise, really great,